libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1447,11 +1447,21 @@ void ExpressionCompiler::appendOrdinaryBinaryOperatorCode(Token::Value _operator
 
 void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Type const& _type)
 {
-	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
-	bool const c_isSigned = type.isSigned();
+//	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+	auto const* intType = dynamic_cast<IntegerType const*>(&_type);
+	auto const* fixedType = dynamic_cast<FixedPointType const*>(&_type);
 
-	if (_type.category() == Type::Category::FixedPoint)
```

Comment:
Please add `solAssert(intType || fixedType, "Expected either integer or fixed point type.");`

Reply:
This was done.